_docs/rfc/scaffolding_new_modules.md

```diff
@@ -0,0 +1,402 @@
+
+- `terragrunt new environment`, which will add a new environment to an existing region. The inputs for this template
+  should be:
+    - Location: which live region/account folder should the environment be added to?
+    - Stack: which stack should be used? This assumes you have different kinds of stacks in your environment, where a
+      stack is a set of components.
+
+- `terragrunt new region`, which will add a new region to an existing account. The inputs for this template should be:
+    - Account: which live account folder should the region be added to? (NOTE: `account` may not be the right
+      terminology here, as it does not mesh with GCP where you have Projects).
+
+- `terragrunt new account`, which will add a new account. The inputs for this template should be:
+    - Remote state info: the minimum information necessary to setup the remote state config.
+
+- `terragrunt new wrapper-module`, which will add a new wrapper module (what you have in `infrastructure-modules`). Note
+  that as such, this will be rendered into `infrastructure-modules`, as opposed to `infrastructure-live` like the other
+  sub commands. The inputs for this template should be:
+    - Module name: what to call this module.
+    - Library Module source: what the underlying library module you are basing the wrapper on.
+
+- `terragrunt new unicorn` (__PLACEHOLDER: I don't know what to call this__. Ideas: `other`, `arbitrary`, `config`),
+  which will support adding any arbitrary template. The inputs depend on the selected template, but at the very least
+  includes:
+    - Template source: where does the template to use live?
+
+### Setup
+
+`terragrunt new` depends on a repository of templates. These templates can practically live anywhere, but for the
+purposes of versioning, it should live outside of `infrastructure-live` and `infrastructure-modules`. For the purposes
+of this example, we will assume all the templates live in a repository called `infrastructure-templates`.
+
+To support each of the scenario above, `terragrunt` will need to be configured to know which subtemplates should be used
+for each of the scenarios, in addition to knowing where the `infrastructure-live` and `infrastructure-modules`
+repositories live. This should be configured in a new configuration file for Terragrunt, which will be autogenerated
+from an interactive setup process the first time you run `terragrunt new`.
+
+The following configuration options will be required:
+
+- `infrastructure_live_repo`: Git URL of the `infrastructure-live` repository.
+- `infrastructure_modules_repo`: Git URL of the `infrastructure-modules` repository.
+- `infrastructure_templates_repo`: Git URL of the `infrastructure-templates` repository. If you follow the convention,
+  this is the only necessary config to discover the templates for each of the scenarios listed above.
+- `new_environment_template_source`: Git URL OR local path to the template source for the `new environment` scenario.
```

Comment:
I was thinking it might be cleaner to have this work by convention. That is, your `infrastructure-templates` repo should have a folder called `scaffolding`, and for each folder `scaffolding/xxx`, you can run `terragrunt new xxx`. Therefore, the default `infrastructure-templates` repo would be at `github.com/gruntwork-io/infrastructure-templates` and it will have `scaffolding/environment`, `scaffolding/region`, `scaffolding/account`, and so on. If a user overrides `infrastructure-templates`, they can replace any of those templates with any names / scaffolds they choose.

Reply:
Yes that is exactly what I was thinking. See:

> This can be used to override the location of the template, as an alternative to the convention followed in infrastructure-templates, or even source from a completely different repository.

at the end of the sentence. This interface is for situations where you don't want to follow convention, like if you wanted a repo per template.